index.md

```diff
@@ -23,6 +27,52 @@ https://github.com/kubernetes-incubator/cri-o
 
 ![CRI-O architecture](assets/images/architecture.png)
 
+The architectural components are as follows:
+
+- Pods live in a `cgroups` slice and hold shared IPC, net and PID `namespaces`.
+- The root filesystem for a container is generated by the containers/storage library when CRI CreateContainer/RunPodSandbox APIs are called.
+- Each container has a monitoring process, `conmon` that receives the master pseudo-terminal `pty`, copies data back and forth between master/slave `pty` pair, handles logging for the container and records the exit code for the container process.
+- The CRI Image API is implemented using the `containers/image` library.
+- Networking for the pod is setup through use of [CNI](https://github.com/containernetworking/cni), so any CNI plugin can be used with CRI-O.
+
+CRI-O is made up of several components that are found in different GitHub repositories.
+Let’s take a look at the components.  
+
+### OCI compatible runtimes
+
+CRI-O supports any [OCI compatible runtime](https://github.com/opencontainers/runtime-tools).
+We test with runc and Clear Containers today.
+
+### Storage
+
+The [containers/storage](https://github.com/containers/storage) lib is used for managing layers
+and creating root file-systems for the containers in a pod: `Overlayfs`, `devicemapper`, `AUFS` and `btrfs`
+are implemented, with `Overlayfs` as the default driver.
+
+There is also support for network based file system images (NFS, Gluster, Cefs) on the way.
+
+### Container images
+
+The [containers/image](https://github.com/containers/image) lib is used for pulling images from registries.
+Currently, it supports Docker [schema 2/version 1](https://docs.docker.com/registry/spec/manifest-v2-1/)
+as well as [schema 2/version 2](https://docs.docker.com/registry/spec/manifest-v2-2/).
+It also passes all Docker and Kubernetes tests.
+
+### Networking
+The Container Network Interface [CNI](https://github.com/containernetworking/cni) is used for setting up networking for the pods.
+Various CNI plugins such as Flannel, Weave and OpenShift-SDN have been tested with CRI-O and are working as expected.
+
+### Monitoring
+
+[conmon](https://github.com/kubernetes-incubator/cri-o/tree/master/conmon) is a utility within CRI-O that is used to
+monitor the containers, handle logging from the container process, serve attach clients and detects Out Of Memory (OOM)
+situations.
+
+### Security
+
+Container security separation policies are provided by a series of tools including SELinux, Capabilities, seccomp,
+and other security separation policies as specified in the OCI Specification.
```

Comment:
So... basically Linux. Should we call this out? A "Containers are Linux" kind of thing. Or... "Workload separation" == "leveraging existing Linux features".